keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -638,24 +640,107 @@ always dispatched immediately.  Following is how the other requests
+sum_assured  = sum[priority levels k] assuredConcurrencyShares(k)
+```
+
+where SCL is the apiserver's concurrency limit.
+
+Borrowing is further limited by a practical consideration: we do not
+want a global mutex covering all dispatching.  Aside from borrowing,
+dispatching from one priority level is done independently from
+dispatching at another.  Borrowing is allowed in just one direction
+(higher may borrow from lower, and lower does not actively lend to
+higher) and in a very limited quantity: an attempt to dispatch for one
+priority level will consider borrowing from just one other priority
+level (if there is any lower priority level at all).
+
+A request can be dispatched exactly at a non-exempt priority level
+when either there are no requests executing at that priority level or
+the number of seats needed by that request is no greater than the
+number of unused seats at that priority level.  The number of unused
+seats at a given priority level is that level's
+NominalConcurrencyLimit minus the number of seats used by requests
+executing at that priority level (dispatched from that priority level
+and higher ones).
+
+There are two sorts of times when dispatching to a non-empty priority
+level is considered: when a request arrives, and when a request
+releases the seats it was occupying (which is not the same as when the
+request finishes from the client's point of view, see below about
+WATCH requests).
+
+At each of these sorts of moments, as many requests are dispatched
+exactly at the same priority level as possible.  The next request to
+consider dispatching is chosen by using the Fair Queuing for Server
+Requests algorithm below to choose a queue at that priority level, and
+the request at the head of that queue is considered.  If (a) no
+requests can be dispatched exactly at that priority level at that
+moment, (b) there are non-empty queues at that level, and (c) there
+are lower non-exempt priority levels, then the request at the head of
+the chosen queue is considered for dispatch at one of the lower
+priority levels.  The particular lower priority level considered is
+drawn at random from the lower ones, in proportion to their
+LendableConcurrencyLimit (we use a static value so that the drawing
+can be done without acquiring mutexes).  The request is executed at
+the chosen lower level (occupying some of its seats) if the request
+can be dispatched exactly at that level according to the rule above.
+
+The following table shows the current default non-exempt priority
+levels and a proposal for their new configuration.
+
+| Name | Assured Shares | Proposed Lendable Shares | Proposed Priority |
+| ---- | -------------: | -----------------------: | ----------------: |
+| leader-election |  10 |   0 |   200 |
+| node-high       |  40 |  10 |   400 |
+| system          |  30 |  10 |   600 |
+| workload-high   |  40 |  20 |  1000 |
+| workload-low    | 100 |  90 |  8000 |
```

Comment:
Workload low is relatively widely used by different controllers. So I think it should have higher priority. More like 2000 in my opinion. And potentially also lower lendable shares?

Reply:
Revised and added an explanation.  A new thought occurred during revision: what happens when reading a pre-existing v1beta2.PriorityLevelConfiguration object that was stored before the new fields were introduced?